src/form/Form.js

```diff
@@ -0,0 +1,170 @@
+"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["d3", "../common/HTMLWidget", "./Input", "css!./Form"], factory);
+    } else {
+        root.form_Form = factory(root.d3, root.common_HTMLWidget,root.form_Input);
+    }
+}(this, function (d3,HTMLWidget,Input) {
+    function Form() {
+        HTMLWidget.call(this);
+        this._class = "form_Form";
+
+        this._tag = "form";
+    }
+    Form.prototype = Object.create(HTMLWidget.prototype);
+    Form.prototype.publish("validate", true, "boolean", "Enable/Disable input validation");
+    Form.prototype.publish("inputs", [], "widgetArray", "Array of input widgets");
+    Form.prototype.publish("controls", [], "widgetArray", "Array of input widgets to control form functionality");
+
+    Form.prototype.testData = function () {
+        this
+            .inputs([
+                new Input()
+                    .id("textbox-test")
+                    .label("Alphanumeric")
+                    .type("textbox")
+                    .validate("^[A-Za-z0-9]+$")
+                    .value("SomeString123"),
+                new Input()
+                    .id("textbox-test")
```

Comment:
ID's should be unique - if you don't set the ID then it will be assigned a unique one automatically.